lib/src/main/java/com/marcinorlowski/bettertextinputlayout/BetterTextInputLayout.java

```diff
@@ -21,24 +21,37 @@
 
 public class BetterTextInputLayout extends android.support.design.widget.TextInputLayout {
 
+	private Field passwordToggleViewField;
+	private Method passwordVisibilityToggleRequestedMethod;
+	
 	public BetterTextInputLayout(Context context) {
 		super(context, null);
 	}
 
 	public BetterTextInputLayout(Context context, AttributeSet attrs) {
-		super(context, attrs, 0);
+		super(context, attrs);
+		init(context, attrs);
 	}
 
 	public BetterTextInputLayout(Context context, AttributeSet attrs, int defStyleAttr) {
 		// Can't call through to super(Context, AttributeSet, int) since it doesn't exist on API 10
 		super(context, attrs);
-
+		init(context, attrs);
+	}
+	
+	private void init(Context context, AttributeSet attrs) {
 		final TintTypedArray a = TintTypedArray.obtainStyledAttributes(context, attrs,
 				R.styleable.TextInputLayout, defStyleAttr, R.style.Widget_Design_TextInputLayout);
 
 		setPasswordVisibilityToggleEnabled(a.getBoolean(R.styleable.TextInputLayout_passwordToggleEnabled, false));
 
 		a.recycle();
+		
+		Class<?> superCls = getClass().getSuperclass();

```

Comment:
Would be useful to wrap this block in `try/catch` and at least do `printStackTrace()` in `catch` so it'd cast the light in case of any troubles